### ISSUE TITLE
CompilerVersionPartWriter::Write crashes with E_POINTER when git SHA version is not available

### ISSUE THREAD

**Issue by LukasBanana** [microsoft/DirectXShaderCompiler#3435]
We build DXC for UnrealEngine without the git version as this simplifies our build workflow in P4V.
The recently introduced class `CompilerVersionPartWriter` writes the commit SHA version without checking for an empty string:
https://github.com/microsoft/DirectXShaderCompiler/blob/bae509a7c5d7b49eabec46ee63167a8d3d0c3712/tools/clang/tools/dxcompiler/dxcompilerobj.cpp#L149

However, if git is not part of the build the initialization of that string is omitted via condition `SUCCEEDED(pVersionInfo->QueryInterface(&pVersionInfo2))`:
https://github.com/microsoft/DirectXShaderCompiler/blob/bae509a7c5d7b49eabec46ee63167a8d3d0c3712/tools/clang/tools/dxcompiler/dxcompilerobj.cpp#L113

So either an empty string should be initialized, or this particular line should handle an empty string differently, or the `IStream::Write` function should accept null pointers as long as the size parameter `cb` is also zero.

Thanks and kind regards
Lukas

**Comment by jaebaek** [comment]
@LukasBanana Thank you for reporting this issue. I will take a look.

**Comment by ehsannas** [comment]
@pow2clk @adam-yang 
This was added via https://github.com/microsoft/DirectXShaderCompiler/commit/2b8ef5b57e4ab0c0a18f3d619a2cbffccdc7fa4d. I had incorrectly thought this was some code that I had introduced back in the day. Jaebaek and I are not sure how to hit this code path (regular dxil/spirv compile didn't hit it), so we don't know how to test a fix for it properly. I'll assign it back to you. Thanks
